src/server/routes/actions.ts

```diff
@@ -26,19 +33,90 @@ router.get<unknown, BalanceResponse>("/balance", (_, res) => {
     });
 });
 
+const missingParameterError = (
+  res: Response<DripErrorResponse>,
+  parameter: keyof BotDripRequestType | keyof FaucetRequestType,
+): void => {
+  res.status(400).send({ error: `Missing parameter: '${parameter}'` });
+};
+
+const addressMiddleware = (
+  req: Request<unknown, DripResponse, Partial<DripRequestType>>,
+  res: Response,
+  next: NextFunction,
+): void => {
+  if (!req.body.address) {
+    return missingParameterError(res, "address");
+  }
+  next();
+};
+
 const dripRequestHandler = new DripRequestHandler(actions, storage, recaptchaService);
 
+router.post<unknown, DripResponse, FaucetRequestType>("/drip/web", addressMiddleware, async (req, res) => {
```

Comment:
One last nitpicking on types from my side :D

You check the `address` with `addressMiddleware`, but not the rest.
So how about something like:

```suggestion
router.post<unknown, DripResponse, Partial<FaucetRequestType> & Pick<FaucetRequestType, 'address'>>("/drip/web", addressMiddleware, async (req, res) => {
```

Then the TS types will reflect reality - that `address` is defined, and about the rest we're not sure, and TS will make sure we handle all missing parameters.